src/main/java/duke/ui/SilentUi.java

```diff
@@ -0,0 +1,31 @@
+package duke.ui;
+
+import java.util.Scanner;
+
+/**
+ * Ui used internally that does not print any output.
```

Comment:
Interesting implementation, maybe the explanation of the class could be a bit more in-depth.